Sources/AsyncAlgorithms/AsyncDeferredSequence.swift

```diff
@@ -0,0 +1,77 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Async Algorithms open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+@inlinable
+public func deferred<Base: AsyncSequence>(_ createSequence: @escaping @Sendable () async -> Base) -> AsyncDeferredSequence<Base> {
+  AsyncDeferredSequence(createSequence)
+}
+
+@inlinable
+public func deferred<Base: AsyncSequence>(_ createSequence: @autoclosure @escaping @Sendable () -> Base) -> AsyncDeferredSequence<Base> {
+  AsyncDeferredSequence(createSequence)
+}
+
+public struct AsyncDeferredSequence<Base: AsyncSequence> {
+  
+  @usableFromInline
+  let createSequence: @Sendable () async -> Base
+  
+  @usableFromInline
```

Comment:
if it is frozen the initializers should then be `@inlinable`. (including the iterator too)